ruma-events/src/lib.rs

```diff
@@ -220,6 +224,13 @@ pub trait EventContent: Sized + Serialize {
 
     /// Constructs the given event content.
     fn from_parts(event_type: &str, content: Box<RawJsonValue>) -> Result<Self, serde_json::Error>;
+
+    /// Constructs the redacted event content.
+    ///
+    /// If called for anything but "empty" redacted content this will error.
+    fn redacted(_event_type: &str) -> Result<Self, serde_json::Error> {
+        Err(serde::de::Error::custom("this event is not redacted"))
+    }
```

Comment:
This solution with the default implementation is quite interesting, however I think we should just have more traits. Yeah, we already have lots of them, but the reason for that is that we don't want to allow invalid combinations of generic event type & event content type. So, unless I'm missing something, we need a `Redacted` version of every existing `EventContent` trait, with the same supertrait hierarchy plus `RedactedEventContent` having a supertrait of `EventContent`. This method, and `has_fields` I proposed earlier (or something equivalent) would then move to `RedactedEventContent`.